pkg/validation/compatibility.go

```diff
@@ -0,0 +1,153 @@
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/blang/semver"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type repository struct {
+	Id  string `json:"id"`
+	Url string `json:"url"`
+}
+
+type pluginMetadata struct {
+	PluginId    string    `json:"id"`
+	Description string    `json:"description"`
+	Provider    string    `json:"provider"`
+	Releases    []release `json:"releases"`
+}
+
+type release struct {
+	Version       string    `json:"version"`
+	Date          string    `json:"date"`
+	Requires      string    `json:"requires"`
+	Compatibility *platform `json:"compatibility"`
+	Hash          string    `json:"sha512sum"`
+	State         string    `json:"state"`
+	Url           string    `json:"url"`
+}
+type platform struct {
+	SpinnakerVersions []string `json:"spinnaker"`
+}
+
+type plugin struct {
+	Id      string
+	Version string
+}
+
+type CompatibilityResult struct {
+	PluginId      string
+	PluginVersion string
+	IsCompatible  bool
+	Reason        string
+}
+
+func ArePluginsCompatible(spinnakerVersion string, plugins []plugin, repo []string) []CompatibilityResult {
```

Comment:
The return types here should be `(error, []CompatibilityResult)`, and instead of logging errors in this function you should return the error to the caller.